src/main/java/org/jabref/gui/maintable/OpenUrlAction.java

```diff
@@ -0,0 +1,73 @@
+package org.jabref.gui.maintable;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import javafx.beans.binding.BooleanBinding;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.ActionHelper;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.desktop.JabRefDesktop;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+
+public class OpenUrlAction extends SimpleCommand {
+
+    private final DialogService dialogService;
+    private final StateManager stateManager;
+
+    public OpenUrlAction(DialogService dialogService, StateManager stateManager) {
+        this.dialogService = dialogService;
+        this.stateManager = stateManager;
+
+        BooleanBinding fieldIsSet = ActionHelper.isFieldSetForSelectedEntry(StandardField.URL, stateManager)
+                                                .or(ActionHelper.isFieldSetForSelectedEntry(StandardField.DOI, stateManager)
```

Comment:
This can be written a bit more elegantly now using the `isAnyFieldSet` method, right?